openedx_events/content_authoring/data.py

```diff
@@ -0,0 +1,64 @@
+"""
+Data attributes for events within the architecture subdomain ``content_authoring``.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+
+The attributes for the events come from the CourseDetailView in the LMS, with some unused fields removed
+(see deprecation proposal at https://github.com/openedx/public-engineering/issues/160)
+"""
+from datetime import datetime
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+@attr.s(frozen=True)
+class CourseScheduleData:
+    """
+    Data describing course scheduling.
+
+    Arguments:
+        start (datetime): course start date
+        end (datetime): course end date
+        pacing (str): 'instructor' or 'self'
+        enrollment_start (datetime): start of course enrollment (optional)
+        enrollment_end (datetime): end of course enrollment (optional)
+    """
+
+    start = attr.ib(type=datetime)
+    end = attr.ib(type=datetime)
+    pacing = attr.ib(type=str)
+    enrollment_start = attr.ib(type=datetime, default=None)
+    enrollment_end = attr.ib(type=datetime, default=None)
+
+
+@attr.s(frozen=True)
+class CourseCatalogData:
+    """
+    Data needed for a course catalog entry.
+
+    Arguments:
+        course_key (CourseKey): identifier of the Course object.
+        name (str): course name
+        org (str): course organization identifier
+        number (str): course number
+        short_description (str): one- or two-sentence course description (optional)
+        effort (str): estimated level of effort in hours per week (optional). Kept as a str to align with the lms model.
+        schedule_data (CourseScheduleData): scheduling information for the course
+        hidden (bool): whether the course is hidden from search
+        invitation_only (bool): whether the course requires an invitation to enroll
+    """
+
+    # basic identifiers
+    course_key = attr.ib(type=CourseKey)
+    name = attr.ib(type=str)
+    number = attr.ib(type=str)
+    org = attr.ib(type=str)
+
+    # additional marketing information
+    short_description = attr.ib(type=str, default=None)
+    effort = attr.ib(type=str, default=None)
```

Comment:
Is there a standard unit of "effort"? Or is this a freeform description left up to the course team?

Reply:
It's supposed to be hours per week, but the field in the lms is just freeform text

Reply:
Got it. Makes sense.